calabash-cucumber/lib/calabash-cucumber/usage_tracker.rb

```diff
@@ -56,6 +57,21 @@ def post_usage_async
       private
 
       # @!visibility private
+      def preferences
+        @preferences ||= Calabash::Cucumber::Preferences.new
+      end
+
+      # @!visibility private
+      def user_id
+        @user_id ||= preferences.user_id
+      end
+
+      # @!visibility private
+      def allowed_to_track
+        @allowed_to_track ||= preferences.usage_tracking
+      end
+
+      # @!visibility private
       def self.track_usage?
         @@track_usage && !self.xtc?
```

Comment:
On that note, what happens if, for some reason, the user changes their prefs file mid-run? Probably an edge case, but it might be easy enough to re-initialize those variables every time you want to send session info (instead of storing their values in ivars).